command_line/fp3.py

```diff
@@ -0,0 +1,578 @@
+import procrunner
+import sys
+import os
+import glob
+import copy
+import concurrent.futures
+import logging
+import time
+
+logger = logging.getLogger("fp3")
+
+from iotbx import phil
+from dxtbx.model.experiment_list import ExperimentList, ExperimentListFactory
+from dials_scratch.fp3 import even_blocks, index_blocks, format_phil_include
+from dials_scratch.fp3 import nproc, combine_reflections, combine_experiments
+from dials_scratch.fp3 import find_setup_script
+
+scope = phil.parse(
+    """
+find_spots {
+  include scope dials.command_line.find_spots.phil_scope
+}
+index {
+  include scope dials.command_line.index.phil_scope
+}
+refine {
+  include scope dials.command_line.refine.phil_scope
+}
+integrate {
+  include scope dials.command_line.integrate.phil_scope
+}
+symmetry {
+  include scope dials.command_line.symmetry.phil_scope
+}
+scale {
+  include scope dials.command_line.scale.phil_scope
+}
+resolution {
+  include scope dials.command_line.estimate_resolution.phil_scope
+}
+parallelism = *process drmaa
+  .type = choice
+max_workers = 1
+  .type = int
+  .help = "Maximum number of concurrent workers"
+worker_nproc = 1
+  .type = int
+  .help = "Number of processes for each worker"
+log_level = CRITICAL ERROR WARNING *INFO DEBUG
+  .type = choice
+""",
+    process_includes=True,
+)
+
+
+class FP3:
+    def __init__(self, filenames, params):
+        self._experiment = ExperimentListFactory.from_filenames(filenames)
+
+        # parse PHIL parameters
+        clai = scope.command_line_argument_interpreter()
+        self._working = scope.fetch(clai.process_and_fetch(params))
+        self._params = self._working.extract()
+        self._debug = self._params.log_level == "DEBUG"
+
+        # configure logging
+        logging.basicConfig(
+            filename="dials.fp3.log", filemode="w", format="%(message)s"
+        )
+        logger.addHandler(logging.StreamHandler(sys.stdout))
```

Comment:
It looks like the various logging things here are doing stuff that `dials.util.log.config` handles. It might be better to use that directly, and hence get the `ColoredFormatter` etc?

Reply:
Could do, will leave in the queue as not familiar with `dials.util.log.config`

Is in 3.1 release series though so not against this.